tests/CMakeLists.txt

```diff
@@ -243,127 +252,147 @@ FOREACH(_test ${_tests})
+
+      ADD_CUSTOM_COMMAND(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.diff
+	COMMAND
+	    ${CMAKE_CURRENT_SOURCE_DIR}/cmake/diff_test.sh 
+		${TEST_DIFF}
+		${testname}/${_name}
+		${CMAKE_CURRENT_SOURCE_DIR}
+		${CMAKE_CURRENT_BINARY_DIR}
+	DEPENDS ${_ref_file}
+	        ${_run_file})
+
+      # Add the target for this output file to the dependencies of this
+      # test. Note that a target may not contain "/" but outputs can be
+      # in subdirectories (for example solution/solution...), so replace them.
+      STRING(REPLACE "/" "-" _target_name ${testname}.${_name}.diff)
+      ADD_CUSTOM_TARGET(${_target_name}
+	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/output-${testname}/${_name}.diff)
+      ADD_DEPENDENCIES(tests.${testname} ${_target_name})
+
+    ENDFOREACH()
 
     # for each of the output files saved in the source directory for
     # this test, create targets that
     # - create a 'output.notime' file
     # - diff this file with the saved one and save the result
-    FILE(GLOB_RECURSE _outputs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/
-	${CMAKE_CURRENT_SOURCE_DIR}/${_test}/[a-zA-Z0-9]*)
 
     FOREACH(_output ${_outputs})
 
-      GET_FILENAME_COMPONENT(ASPECT_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR} PATH)
-
-      # create the output.notime targets for each of the output files,
-      # and also create a similar file for all of the input files (this
-      # avoids the need to strip timing information from files we may
-      # copy from tests we have just run to the directory where we save
-      # these files
-      #
-      # do the same with the second egrep command to strip out the status
-      # message at the top of the run
-      ADD_CUSTOM_COMMAND(
-	OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime
-	COMMAND
-	  cat ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/${_output}
-	  | egrep -v '^\\|'
-	  | egrep -v '^--'
-	  > ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime
-	DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/${_output}
-      )
 
       # The screen-output target needs to normalize the .cmp.notime
       # tests. They need to exist first, so add an explicit dependency. To do
       # that, we first create a target to the command above (ADD_DEPENDENCIES
       # can only deal with targets, not commands). Finally, we need to remove "/"
       # (illegal in target names).
-      STRING(REPLACE "/" "-" _target_name_copy_output make-${_test}-${_output}.cmp.notime)
-      ADD_CUSTOM_TARGET(${_target_name_copy_output}
-	DEPENDS	${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime)
-      ADD_DEPENDENCIES(${_test}.normalize ${_target_name_copy_output})
+#      STRING(REPLACE "/" "-" _target_name_copy_output make-${_test}-${_output}.cmp.notime)
+ #     ADD_CUSTOM_TARGET(${_target_name_copy_output}#
+#	DEPENDS	${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime)
+ #     ADD_DEPENDENCIES(${_test}.normalize ${_target_name_copy_output})
```

Comment:
what's happening here with these commented out things? same in the following lines?